app/services/katello/pulp3/smart_proxy_repository.rb

```diff
@@ -33,6 +33,38 @@ def delete_orphaned_repositories_for_mirror_proxies
           ::Katello::Pulp3::Repository.new(nil, smart_proxy).repositories_api.delete(repo._href)
         end
       end
+
+      def pulp3_enabled_repo_types(content_types = SETTINGS[:katello][:content_types].keys)
+        repo_types = content_types.collect do |content_type|
+          Katello::RepositoryTypeManager.find(content_type)
+        end
+
+        repo_types.select { |type| type.try(:pulp3_service_class) }
+      end
+
+      def delete_orphaned_distributions_for_mirror_proxies(repo_types = nil)
+        repo_types ||= pulp3_enabled_repo_types
+        repo_types.each do |repo_type|
+          pulp3_repo = repo_type.pulp3_service_class.new(nil, smart_proxy)
+          orphan_distribution_hrefs = ::Katello::Pulp3::Repository.orphan_distributions(pulp3_repo, {})
+          orphan_distribution_hrefs.each do |distribution_href|
+            pulp3_repo.delete_distribution(distribution_href)
```

Comment:
and maybe to the same token here, if this was a static method (like orphan_distributions), which took in a smart_proxy and distribution_href, it could handle that.  Although maybe there's a way to structure some of this stuff?  But if not lets at least hide it as best we can.